packages/vega-parser/src/parsers/guides/guide-util.js

```diff
@@ -86,3 +87,80 @@ export function extendOffset(value, offset) {
     : !isObject(value) ? { value, offset }
     : { ...value, offset: extendOffset(value.offset, offset) };
 }
+
+const DISCRETE_SCALES = new Set(['ordinal', 'band', 'point']);
+
+function formatType(_, scaleType) {
+  const formatType = _('formatType');
+
+  if (formatType) {
+    return formatType;
+  }
+  if (scaleType === Time) {
+    return 'time'
+  }
+  if (scaleType === UTC) {
+    return 'utc'
+  }
```

Comment:
Could this method return `undefined`? If so, what happens subsequently? Should there be an error condition here?

Reply:
It's now

```
function formatType(_, scaleType) {
  return _('formatType') || scaleType;
}
```